xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -1180,10 +1181,11 @@ bool CDecoder::Open(AVCodecContext* avctx, AVCodecContext* mainctx, enum AVPixel
       // a driver may use multi-thread decoding internally
       m_refs += CServiceBroker::GetCPUInfo()->GetCPUCount();
     }
-    else
-      m_refs += CServiceBroker::GetCPUInfo()->GetCPUCount() / 2;
```

Comment:
xbox only has memory for 16 surfaces (according Kodi comments in code).

Current code uses 8 initial surfaces + 8 in case of HEVC = 16

If we add more surfaces by (CPU cores / 2) code not works in xbox for sure.

Since now is fixed 16 surfaces for xbox (and no exist possible/legal HEVC stream with more of 8 refs) is supposed all streams should work in xbox with 16 surfaces (anyways not possible add more surfaces on xbox).